client/app/utils/numbers.js

```diff
@@ -1,23 +1,121 @@
+import _ from 'lodash';
+import Immutable from 'immutable';
+import numbro from 'numbro';
+import { t } from './i18n';
+import currencies from '../assets/json/currency_iso.json';
+
 const MINIMUM_DISTANCE = 0.1;
 const PRECISION = 2;
 const DECIMAL_BASE = 10;
 
+// Locales, which are not yet supported by numbro, are not included.
+// So, these languages default to en-US formatting.
+// el-GR, es-CL, ca-ES, hr-HR, id-ID, is-IS, km-KH, ms-MY,
+// ro-RO, sw-KE, vi-VN, bg-BG, mn-MN, zh-HK, ka-GE, sl-SI, ti-ER
+const DEFAULT_LOCALE = 'en-US';
+const KNOWN_LOCALES = [
+  'en-US',
+  'da-DK',
+  'de-DE',
+  'en-AU',
+  'en-GB',
+  'es-ES',
+  'fi-FI',
+  'fr-FR',
+  'fr-CA',
+  'it-IT',
+  'ja-JP',
+  'nb-NO',
+  'nl-NL',
+  'pt-BR',
+  'pt-PT',
+  'ru-RU',
+  'sv-SE',
+  'tr-TR',
+  'zh-CN',
+  'en-NZ',
+  'et-EE',
+  'pl-PL',
+  'hu-HU',
+  'cs-CZ',
+  'th-TH',
+  'zh-TW',
+  'sk-SK',
+];
+
+const initializeNumbro = _.memoize((n, locale) => {
+  if (locale && KNOWN_LOCALES.includes(locale) && locale !== 'en-US') {
+    n.culture(locale, require(`numbro/languages/${locale}`));
+  } else {
+    KNOWN_LOCALES.forEach((localeCode) => {
+      if (localeCode !== 'en-US') {
+        n.culture(localeCode, require(`numbro/languages/${localeCode}`));
+      }
+    });
+  }
+  return n;
+}, (n, locale) => (locale ? locale : 'all'));
+
+const localizeNumbro = function localizeNumbro(locale) {
+
+  // Ensure that numbro has required all the localization languages and select correct locale
+  if (KNOWN_LOCALES.includes(locale)) {
+    const isServer = typeof window === 'undefined';
+    const numbroMultiLang = initializeNumbro(numbro, isServer ? null : locale);
+    numbroMultiLang.culture(locale);
+    return numbroMultiLang;
+  }
+  numbro.culture(DEFAULT_LOCALE);
+  return numbro;
+};
+
+const translateDistanceUnit = function translateDistanceUnit(unit) {
+  switch(unit) {
+    case ':km':
+      return t('web.utils.km');
+    case ':miles':
+      return t('web.utils.mi');
+    default:
+      throw new Error('Unknown distance unit');
+  }
+}
+
 const sigFigs = function sigFigs(n, sig) {
   return parseFloat(n.toPrecision(sig));
 };
 
-const formatDistance = function formatDistance(distance, unit, precision = PRECISION, minimumDistance = MINIMUM_DISTANCE) {
+const formatDistance = function formatDistance(distance, locale = DEFAULT_LOCALE, precision = PRECISION, minimumDistance = MINIMUM_DISTANCE) {
   if (distance == null) {
     return null;
   }
 
-  const precisionWithCloseBy = (distance < 1) ? 1 : precision;
-  return (distance < minimumDistance) ?
-    `< 0.1 ${unit}` :
-    `${sigFigs(distance, precisionWithCloseBy)} ${unit}`;
+  const localizedNumbro = localizeNumbro(locale);
+  const dist = localizedNumbro(sigFigs(distance.value, precision));
+  const localizedUnit = translateDistanceUnit(distance.unit);
+  const formatted = (distance.value < 1) ? dist.format('0,0.0') : dist.format('0,0');
```

Comment:
Nope, those are just placeholders for locale specific rules. So, that *'0,0.0'* should be read as "use thousand separator and decimal separator [according to installed culture rules]"